__tests__/articles.test.js

```diff
@@ -122,3 +121,88 @@ describe("GET /api/articles/:article_id/comments", () => {
       });
   });
 });
+
+describe("POST: /api/articles/:article_id/comments", () => {
+  test("201: returns the successfully posted comment", () => {
+    const testComment = {
+      comment: {
+        username: "butter_bridge",
+        body: "Bemused of Basingstoke. What is this guy on?",
+      },
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(testComment)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("body", expect.any(String));
+        expect(comment).toHaveProperty("article_id", expect.any(Number));
+        expect(comment).toHaveProperty("author", expect.any(String));
+        expect(comment).toHaveProperty("votes", 0);
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+      });
+  });
+  test("400: returns error if article_id doesn't exist", () => {
+    const testComment = {
+      comment: {
+        username: "butter_bridge",
+        body: "Bemused of Basingstoke. What is this guy on?",
+      },
+    };
+    return request(app)
+      .post("/api/articles/14/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("400: returns error if username doesn't exist", () => {
+    const testComment = {
+      comment: {
+        username: "franklin_d",
+        body: "Bemused of Basingstoke. What is this guy on?",
+      },
+    };
+    return request(app)
+      .post("/api/articles/14/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+
+  test("400: returns error if posted comment contains wrong data types", () => {
+    const testComment = {
+      comment: {
+        username: 25,
+        body: 25,
+      },
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
+  test("400: returns error if posted comment is malformed", () => {
+    const testComment = {
+      comment: {
+        dodgyKey1: "franklin_d",
+        dodgyKey2: "Bemused of Basingstoke. What is this guy on?",
+      },
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
```

Comment:
Could we just add one more 400 test please for invalid ID. I.e. articles/bananas/comments ?